DependencyInjection/FOSUserExtension.php

```diff
@@ -133,6 +132,22 @@ public function load(array $configs, ContainerBuilder $container)
         if (!empty($config['group'])) {
             $this->loadGroups($config['group'], $container, $loader, $config['db_driver']);
         }
+
+        if ($container->hasParameter('fos_user.registration.confirmation.from_email') || $container->hasParameter('fos_user.resetting.email.from_email')) {
+            $mailer = $config['service']['mailer'];
+            $container->setAlias('fos_user.mailer', $mailer);
+            $mailerDefinition = $container->getDefinition($mailer);
```

Comment:
The mailer service I configure can be from a separate bundle than FOSUserBundle (I have a different implementation than the builtin ones, as I don't use swiftmailer in my project). In this case, the DI extension would get an exception about a non-existent service here, as the ContainerBuilder it manipulates starts empty